CS483/CS483/Kartaclysm/Components/ComponentRacer.cpp

```diff
@@ -108,4 +110,17 @@ namespace Kartaclysm
 
 		m_iCurrentPosition = iPosition + 1;
 	}
+
+	void ComponentRacer::SetCurrentTrackPiece(int p_iNewTrackPiece)
+	{
+		m_iCurrentTrackPiece = p_iNewTrackPiece;
```

Comment:
All of the work should be done before this method is called, not after. This would avoid the redundancy of the Track calling this method and having this method call a Track method. You should have this method accept two input parameters, p_iNewTrackPiece and p_iTrackPieceForDistanceCheck, or having a separate method for SetTrackPieceForDistanceCheck.